pkg/apis/azure/validation/credentials.go

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	guidRegex = regexp.MustCompile("^[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}$")
+)
+
+// CredentialMapping maps logical credential field names to their credential keys
+// to enable usage across different credential types (infrastructure, DNS)
+type CredentialMapping struct {
+	// RequiredFields keeps fields that are required
+	RequiredFields map[string]string
+	// OptionalFields keeps fields that are optional
+	OptionalFields map[string]string
+	// GUIDFields keeps fields that must be validated as GUIDs
+	GUIDFields map[string]string
+	// ImmutableFields keeps fields that must not change once set
+	ImmutableFields map[string]string
+}
+
+// ValidateRequiredCredentials validates that all required credential fields are present and non-empty
+func ValidateRequiredCredentials(secret *corev1.Secret, mapping CredentialMapping, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	secretKey := fmt.Sprintf("%s/%s", secret.Namespace, secret.Name)
+
+	for logicalName, dataKey := range mapping.RequiredFields {
+		value, exists := secret.Data[dataKey]
+		if !exists {
+			allErrs = append(allErrs, field.Required(fldPath.Key(dataKey), fmt.Sprintf("missing required field %s in secret %s", logicalName, secretKey)))
```

Comment:
I feel that this may cause some confusion to people. The logical name can be different from the data key and when you report that a required field is missing, then likely people will put fill it as-is